Remote Habits/View/Common/RHBaseViewController.swift

```diff
@@ -0,0 +1,90 @@
+//
+//  BaseViewController.swift
+//  Remote Habits Mobile App
+//
+//  Created by Amandeep Kaur on 25/11/21.
+//
+
+import UIKit
+
+class RHBaseViewController: UIViewController {
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Do any additional setup after loading the view.
+    }
+    
+    // MARK: - Common Methods
+    func addDefaultBackground() {
+        self.view.backgroundColor = RHColor.DefaultBackground
+    }
+    
+    func addLoginBackground() {
+        let imageBg = UIImageView(frame: CGRect(x: 0, y: 0, width: UIScreen.main.bounds.width, height: UIScreen.main.bounds.height))
+        imageBg.image = UIImage(named: RHConstants.kBackground)
+        self.view.addSubview(imageBg)
+        self.view.sendSubviewToBack(imageBg)
+    }
+    
+    @objc func goBack() {
+        self.dismiss(animated: true, completion: nil)
+    }
+
+    func configureNavigationBar(titleColor: UIColor = .black, backgoundColor: UIColor = .white, title: String, hideBack : Bool, showLogo : Bool = false) {
+        if hideBack {
+            self.navigationItem.setHidesBackButton(hideBack, animated: true)
+        }
```

Comment:
```suggestion
        self.navigationItem.setHidesBackButton(hideBack, animated: true)
```

Since you are passing in `hideBack` Bool value anyway, no need for the `if {}`. Also, I imagine a bug could happen if you call `configureNavigationBar` with `hideBack = true` and then call it again with `hideBack = false`.

Reply:
There is an else statement to this `if` condition. In case `hideBack` is `false` then the function will display a back button otherwise hide it. Also, if `hideBack` is false then I didn't want any extra code to be done like updating the value of `setHidesBackButton` so it;s written this way.

Reply:
Ok, I understand that. 

I do believe you will still need to include `self.navigationItem.setHidesBackButton(false, animated: true)` in the `else` condition though, yeah?